src/game/AI/ScriptDevAI/scripts/kalimdor/dustwallow_marsh.cpp

```diff
@@ -1000,6 +1001,175 @@ UnitAI* GetAI_mob_invis_firework_helper(Creature* pCreature)
     return new mob_invis_firework_helper(pCreature);
 }
 
+/*######
+## npc_theramore_practicing_guard
+######*/
+enum
+{
+    NPC_THERAMORE_COMBAT_DUMMY = 4952,
+};
+
+struct npc_theramore_practicing_guardAI : public ScriptedAI
+{
+    npc_theramore_practicing_guardAI(Creature* pCreature) : ScriptedAI(pCreature)
+    {
+        attackableDummy = GetClosestCreatureWithEntry(m_creature, NPC_THERAMORE_COMBAT_DUMMY, 2.f);
+        Reset();
+    }
+
+    uint32 m_attackTimer;
+    uint32 m_breakTimer;
+    uint32 m_sitTimer;
+    bool m_bisAttacking;
+    bool m_bsitDown, m_bstandUp;
+    bool m_binCombatWithPlayer;
+    Creature* attackableDummy;
```

Comment:
Consider storing the creature guid instead of creature pointer. If the attackableDummy gets removed from the world for any reason, you will lose traceability.

If it gets deleted -
if (attackableDummy != nullptr) will return true, 
attackableDummy->AnyMethod will crash your application

Instead, store the creatures guid in ObjectGuid and get it in realtime when you need it from Map.

Reply:
Oh right yeah I forgot to change it, thanks